lib/expected_behavior/acts_as_archival_active_record_methods.rb

```diff
@@ -36,5 +36,23 @@ def is_archival?
 
     end
 
+    module ARRelationMethods
+
+      def archive_all!
+        error_message = "The #{klass} must implement 'act_on_archivals' in order to call `archive_all!`"
+        raise NotImplementedError.new(error_message) unless archival?
+
+        each(&:archive!).tap { reset }
```

Comment:
Just so we're clear, this will create n archive ids so that each archived item is its own tree. Is that intended, or do you want these all to count as a single archive action and have a shared archive id?

Reply:
Thanks for pointing that out -- I think in order to be consistent with the fact that parent-child archivals have the same archive id, let's go ahead and and have any record archived using `archive_all!` share the same archive_id. That change is currently in the PR. I've used the AR Relation's class and an integer representation of the time to generate the hash.